### DISCUSSION TITLE
What is edgeDb

### DISCUSSION THREAD

**Discussion by avin-kavish** [geldata/gel#4021] [General]
Is this actually a database or a query language around a postgresql server?

**Comment by nsidnev** [comment]
It seems to me that both statements can be considered correct :)

EdgeDB uses PostgreSQL as a storage for the data it manages. This is not how storage is used in PostgreSQL itself, but that does not stop the database from being a database, right?

However, EdgeDB uses EdgeQL language for interacting with the user. And in that sense, it can be seen as a frontend to PostgreSQL. But apart from the language itself EdgeDB uses a lot of logic to speed up EdgeQL query processing, such as caches for generated SQL code and various other things, so I don't think EdgeDB can be called just a "query language" around PostgreSQL, because it does a lot of work beyond that.

**Reply by avin-kavish** [reply]
> EdgeDB uses PostgreSQL as a repository for the data it manages.

Do you mean EdgeDb connects to a postgresql server via a driver or does it integrate postgres into the codebase and compile into one binary?

**Reply by nsidnev** [reply]
EdgeDB connects to PostgreSQL via Python [asyncpg](https://github.com/MagicStack/asyncpg) driver
